app/controllers/bookings_controller.rb

```diff
@@ -1,11 +1,22 @@
 class BookingsController < ApplicationController
+  before_action :current_booking, only: [:show, :edit, :update, :destroy]
   def index
+    @bookings = Booking.all
   end
 
   def new
+    @booking = Booking.new
   end
 
   def create
+    @experience = Experience.find(params[:experience_id])
```

Comment:
I would send it to a private method "@experience = Experience.find(params[:experience_id])" and call it again before_action with the method :new and :create